src/docs/development/platform-integration/add-to-app/performance.md

```diff
@@ -0,0 +1,174 @@
+is constructed since the JNI connectors need to reference the Flutter C++
+library. On **iOS**, this happens when the [`FlutterEngine`]({{site.api}}/objcdoc/Classes/FlutterEngine.html)
+is first run, such as via [`runWithEntrypoint:`]({{site.api}}/objcdoc/Classes/FlutterEngine.html#/c:objc(cs)FlutterEngine(im)runWithEntrypoint:).
+
+### Starting the Dart VM
+
+The Dart runtime is responsible for managing Dart memory, concurrency etc. for
+your Dart code. In JIT mode, it's additionally responsible for compiling
+the Dart source code into machine code during runtime.
+
+A single Dart runtime exists per application session on Android and iOS.
+
+A one-time Dart VM start is done when constructing the [`FlutterEngine`]({{site.api}}/javadoc/io/flutter/embedding/engine/FlutterEngine.html)
+for the first time on **Android** and when [running a Dart entrypoint]({{site.api}}/objcdoc/Classes/FlutterEngine.html#/c:objc(cs)FlutterEngine(im)runWithEntrypoint:)
+for the first time on **iOS**.
+
+At this point, your Dart code's [snapshot](https://github.com/dart-lang/sdk/wiki/Snapshots)
+is also loaded into memory from your application's files.
+
+This is a generic process that would also occur if you used the [Dart SDK](https://dart.dev/tools/sdk)
+directly without the Flutter engine.
+
+The Dart VM never shuts down once started.
+
+### Creating and running a Dart Isolate
+
+Once the Dart runtime is initialized, the Flutter engine's usage of the Dart
+runtime is the next step.
+
+This is done by starting a Dart Isolate in the Dart runtime. The Isolate is
+Dart's container for memory and thread. A number of [auxiliary threads](https://github.com/flutter/flutter/wiki/The-Engine-architecture#threading)
+on the host platform is also created at this point to support the Dart Isolate,
+such as offloading GPU handling and image decoding off of the Dart Isolate.
+
+One Dart Isolate exists per `FlutterEngine` instance and multiple Dart Isolates
+can be hosted by the same Dart VM.
+
+On **Android**, this happens when you call [`DartExecutor.executeDartEntrypoint()`]({{site.api}}/javadoc/io/flutter/embedding/engine/dart/DartExecutor.html#executeDartEntrypoint-io.flutter.embedding.engine.dart.DartExecutor.DartEntrypoint-)
+on a `FlutterEngine` instance.
+
+On **iOS**, this happens when you call [`runWithEntrypoint:`]({{site.api}}/objcdoc/Classes/FlutterEngine.html#/c:objc(cs)FlutterEngine(im)runWithEntrypoint:)
+on a `FlutterEngine`.
+
+At this point, your Dart code's selected entrypoint (the `main()` function of
+your Dart library's main.dart file by default) is executed. If you called the
+Flutter function [runApp()]({{site.api}}/flutter/widgets/runApp.html) in your
+`main()` function, your Flutter app or library's widget tree will also be
+created and built. If you need to gate certain functionalities from executing
+in your Flutter code, the `AppLifecycleState.detached` enum value will indicate
```

Comment:
I think it would be more clear if you replaced "gate" with "prevent".

"enum value will indicate" => "enum value indicates"

Reply:
done